kernel/comps/block/src/impl_block_device.rs

```diff
@@ -72,14 +76,18 @@ impl dyn BlockDevice {
     }
 
     /// Synchronously writes one block indicated by the `bid`.
-    pub fn write_block(&self, bid: Bid, frame: &Frame) -> Result<BioStatus, BioEnqueueError> {
+    pub fn write_block(&self, bid: Bid, frame: &AnyFrame) -> Result<BioStatus, BioEnqueueError> {
```

Comment:
In this PR `Frame` (which is `Frame<()>`) appeared 63 times, `AnyFrame`appeared 118 times `CachePage` (which is `Frame<CachePageMeta>`) appeared 12 times.

From my perspective, `AnyFrame` is neither the most encouraged type now nor the most prevalent type in the future:

 - A frame should have it's purpose (anonymous/page cache/buffer) and their accessing behavior will be distinct. Almost all frames will have metadata (anonymous pages will gain it's reverse mapping in the future). So when the metadata of a frame is accessed, `AnyFrame` is not the best option. Metadata will be frequently accessed. For this reason, any internal APIs that are dedicated to one usage (such as VMO, which holds only page cache pages), will eventually turn to typed frame.
 - If `Frame` represents `AnyFrame` rather than `Frame<()>`, the allocation scheme, as discussed in https://github.com/asterinas/asterinas/pull/1287#discussion_r1774375098, makes the API far harder to read and semantically indiscreet. The metadata of an allocated `Frame` can be `()`, but the user cannot tell it from the API. Also, the `Ex` suffix makes less intuitive names. The user cannot tell directly that the extension means attaching metadata other than `()`. The naming dilemma just indicates a bad design.

Also `AnyFrame` is a good name indicating that the handle points to anything that is a frame. The methods of `AnyFrame` can be applied to "any frame".

To sum up, I support leaving the naming as is in this PR.